quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/KubeDiscoveryConfigSourceInterceptor.java

```
@@ -33,36 +30,29 @@ public class KubeDiscoveryConfigSourceInterceptor implements ConfigSourceInterce
 
     private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass().getName());
     private transient KubernetesClient client;
-    private Map<String, String> cache = new ConcurrentHashMap<>();
 
-    private KubeResourceDiscovery kResource;
+    private final transient KubeDiscoveryConfigCache kubeDiscoveryConfigCache;
 
     public KubeDiscoveryConfigSourceInterceptor() {
         logger.debug("Configuring k8s client...");
         this.client = new DefaultKubernetesClient();
-        this.kResource = new KubeResourceDiscovery(client);
+        this.kubeDiscoveryConfigCache = new KubeDiscoveryConfigCache(new KubeResourceDiscovery(client));
     }
 
     @Override
     public ConfigValue getValue(ConfigSourceInterceptorContext context, String s) {
         ConfigValue configValue = context.proceed(s);
-        try {
-            if (isValidURI(configValue)) {
-                String cachedValue = cache.computeIfAbsent(configValue.getName(), k -> kResource
-                        .query(new KubeURI(configValue.getValue())).orElse(null));
-                if (cachedValue != null) {
-                    return configValue.withValue(cachedValue);
-                }
+        if (configValue == null) {
+            return null;
+        } else {
+            try {
+                return kubeDiscoveryConfigCache.get(configValue.getName(), configValue.getValue())
+                        .map(configValue::withValue)
+                        .orElse(configValue);
+            } catch (RuntimeException e) {
+                logger.error("Service Discovery has failed", e);
```

When does this happen? Can we safely swallow this exception?

---

IDK. I didn't change this. @spolti WDYT?

---

IIRC we used this catch to throw any issue that can happen during the discovery, the MicroProfile was suppressing some exception, in other words when something went wrong there would not be exception being thrown.